include/onnxruntime/core/framework/tensor.h

```diff
@@ -58,7 +113,16 @@ using BufferNakedPtr = void*;
 */
 class Tensor final {
  public:
-  Tensor() = default;  // to allow creating vector<Tensor> to support seq(tensor)
+  // to allow creating vector<Tensor> to support seq(tensor)
+  Tensor() noexcept
```

Comment:
Any reason for calling out the members instead of using = default?

Reply:
I was planning to have some additional code there bc of shape and default MLDataType, as implied by the move members. The rules for default initialization of primitive members are somewhat complicated (see Value Initialization vs Default Initialization) and my understanding zero initialization is not always performed although many implementations according to Internet do it. After removing additional code I thought I would better leave it like this.